relay-event-normalization/src/normalize/span/tag_extraction.rs

```diff
@@ -143,6 +146,7 @@ pub(crate) fn extract_span_tags(event: &mut Event, config: &Config) {
     let is_mobile = shared_tags
         .get(&SpanTagKey::Mobile)
         .is_some_and(|v| v.as_str() == "true");
+    let start_type = get_event_start_type(event.clone(), is_mobile);
```

Comment:
```suggestion
    let start_type = is_mobile.then(|| get_event_start_type(&event).flatten();
```

It's important that we do not clone the event, this is very costly. I would also not pass the `is_mobile` flag into the function, that is just hidden control flow the caller can take care of.

Reply:
@Dav1dde so I ended up doing this because I was getting the following error:
```rust
error[E0502]: cannot borrow `event` as immutable because it is also borrowed as mutable
   --> relay-event-normalization/src/normalize/span/tag_extraction.rs:154:37
    |
150 |     let Some(spans) = event.spans.value_mut() else {
    |                       ----------- mutable borrow occurs here
...
154 |     let start_type = is_mobile.then(|| get_event_start_type(&event)).flatten();
    |                                     ^^                       ----- second borrow occurs due to use of `event` in closure
    |                                     |
    |                                     immutable borrow occurs here
155 |     let ttid = timestamp_by_op(spans, "ui.load.initial_display");
    |                                ----- mutable borrow later used here
```

Is there a typical method for getting around this? I think I could store the measurements in a variable and pass that down into `get_event_start_type` but I like that `event` has a `measurements` method that does some of the nice retrieval for me. I'm not so familiar with Rust so please feel free to point me towards a more idiomatic approach

Reply:
Ah sorry, I missed that.

You're correct, we cannot do this, because a part of the event (spans field) is already mutably borrowed and then you pass the entire event to a function, that's why that is not allowed.

The Rust way of fixing this would be just passing the measurements of the event to the function, but in this case since we use the `event.measurement()` method this is not worth the trouble -> there is a good reason to keep it where it is 👍